Documentation/network/concepts/masquerading.rst

```diff
@@ -170,3 +170,10 @@ primary interface address. The latter is then only considered as a catch-all
 fallback, and for the default routes. For these advanced cases the user needs
 to ensure that there are no overlapping destination CIDRs as routes on the
 relevant masquerading interfaces.
+
+With the ``enable-masquerade-to-route-source: "true"`` option, Cilium will, 
+by default, use interfaces listed in the ``devices`` field as the egress masquerade interfaces 
+when ``egress-masquerade-interfaces`` is empty. When ``egress-masquerade-interfaces`` is set, 
+it takes precedence over ``devices`` to choose which network interface should perform masquerading.
+   with ``enable-masquerade-to-route-source: "true"`` option ,
+   you can set egress-masquerade-interfaces as a list such as ``eth1 eth2 eth3``.
```

Comment:
```suggestion
You can set egress-masquerade-interfaces to match multiple interfaces like this: ``eth+``.
```

(drop unnecessary wording; fix indentation complaint from CI; generalize the proposed configuration)

Reply:
Maybe even `eth+ ens+` as an example? This would use both wildcards and the two most common formats for network interface naming schemes.

Reply:
Hmm, it looks like we only supported multiple patterns for masquerade interfaces since this PR: https://github.com/cilium/cilium/pull/36103 . Not sure if `--devices` supported multiple patterns before. We can then either choose to document just the precedence and skip this sentence for backport to v1.15, or include this example for multiple devices and only merge this for v1.17.

Reply:
If we want to backport, maybe the easiest way is we remove this sentence and merge the PR, then you can propose another new PR on top that we don't backport.

Reply:
https://github.com/cilium/cilium/pull/36103 this PR solves the multi-interfaces issue when `enable-masquerade-to-route-source ` is `false`. Multi interfaces are supported since day 1 when we have `masquerade-to-route-source`

Reply:
Ah I see, in that case we can keep this sentence included. :+1:

Reply:
I've just unresolved this comment because the original "suggested change" at the top of the thread still makes sense: The first line is unnecessary because the paragraph is already about that mode; capitalization; indentation needs to be fixed for the linter in CI.

I also think it probably makes sense to recommend `eth+ ens+` to cover the common network interface naming prefixes but I'll defer to you on the exact example to include on this line. It is just an example after all.